### PR TITLE
net: lwm2m: Refactoring of engine, observation

### PR DESCRIPTION
**Pull request by otkulseng**
Created the file lwm2m_observation.c, to include
all code maintaining observations and parameter handling.
Moved the observe_node struct to lwm2m_engine.h as it was needed
in both files.

Signed-off-by: Ola Tangen Kulseng <ola.kulseng@nordicsemi.no>

### CODE REVIEW

File: subsys/net/lib/lwm2m/lwm2m_observation.c [hunk 1]
```diff
@@ -0,0 +1,1438 @@
...
+#if defined(CONFIG_LWM2M_DTLS_SUPPORT)
+#include <zephyr/net/tls_credentials.h>
+#endif
+#if defined(CONFIG_DNS_RESOLVER)
+#include <zephyr/net/dns_resolve.h>
+#endif
+
+#include "lwm2m_object.h"
+#include "lwm2m_engine.h"
+#include "lwm2m_rw_link_format.h"
```
Comment (VeijoPesonen) [inline subsys/net/lib/lwm2m/lwm2m_observation.c:44]: Seems like all the seven content format headers can be removed from here too - similarly to the PR #46876

File: subsys/net/lib/lwm2m/lwm2m_observation.c [hunk 2]
```diff
@@ -0,0 +1,1438 @@
...
+#include <ctype.h>
+#include <errno.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/net/net_ip.h>
+#include <zephyr/net/http_parser_url.h>
+#include <zephyr/net/socket.h>
+#include <zephyr/net/lwm2m.h>
+#if defined(CONFIG_LWM2M_DTLS_SUPPORT)
+#include <zephyr/net/tls_credentials.h>
```
Comment (VeijoPesonen) [inline subsys/net/lib/lwm2m/lwm2m_observation.c:36]: Please drop this one.

File: subsys/net/lib/lwm2m/lwm2m_observation.c [hunk 3]
```diff
@@ -0,0 +1,1438 @@
...
+#endif
+#if defined(CONFIG_DNS_RESOLVER)
+#include <zephyr/net/dns_resolve.h>
```
Comment (VeijoPesonen) [inline subsys/net/lib/lwm2m/lwm2m_observation.c:39]: Please drop this one too.

File: subsys/net/lib/lwm2m/lwm2m_engine.c [hunk 4]
```diff
@@ -141,45 +106,62 @@ static struct lwm2m_ctx *sock_ctx[MAX_POLL_FD];
 static struct pollfd sock_fds[MAX_POLL_FD];
 static int sock_nfds;
 
+/* Resource wrappers */
+struct lwm2m_ctx **lwm2m_sock_ctx(void)
+{
+	return sock_ctx;
+}
+
+int lwm2m_sock_nfds(void)
```
Comment (VeijoPesonen) [inline subsys/net/lib/lwm2m/lwm2m_engine.c:115]: Please add the declaration to the header file. Make the same for all the other functions which are used from _lwm2m_observation.h_.

**Review by VeijoPesonen — CHANGES_REQUESTED** [review]

File: subsys/net/lib/lwm2m/lwm2m_engine.c (hunk 4 above)
Comment (jheiskan81) [reply subsys/net/lib/lwm2m/lwm2m_engine.c:115]: I was wondering same that own header file would be useful in this change.

**Comment by otkulseng** [comment]
This was replaced by: #46976

**Pull request closed without merging**